### DISCUSSION TITLE
BelongsTo doesn't trigger `dependsOn` on value changes

### DISCUSSION THREAD

**Discussion by crynobone** [laravel/nova-issues#4774] [Q&A] [answered]
- Laravel Version: 9.27.0
- Nova Version: 4.13.00 (Silver Surfer)
- PHP Version: 8.0.11
- Database Driver & Version: SqlLite
- Operating System and Version: Windows 10 21H2
- Browser type and version: MS Edge 104.0.1293.63 64-bit and FireFox 91.13.0esr 64bit
- Reproduction Repository: https://github.com/kusab85/TestApp/tree/nova-dependsOn-belongsTo-field

Open Posts page and edit a Post.
Callback function is called **only once** when page is being loaded.
After that, changing value of the User field does not do a thing.
Just look into laravel.log 

Cheers!

UPDATE:
It seems that dependsOn doesn't work at all, no matter what field uses it.
Check 'Post With Dependencies' page in my TestApp

_Originally posted by @kusab85 in https://github.com/laravel/nova-issues/issues/4751#issuecomment-1234355775_

**Comment by crynobone** [comment]
@kusab85 I don't understand why you code it as below:

```php
Text::make('Title')
                ->rules('required')
                ->dependsOn(['user'],
                    function (Text $field, NovaRequest $request, FormData $formData) {
                        logger(get_class($this).": User changed to ".var_export($formData->user, true));
                    }
                )
                ->dependsOn(['status'],
                    function (Text $field, NovaRequest $request, FormData $formData) {
                        logger(get_class($this).": Status changed to ".var_export($formData->status, true));
                    }
                )
                ->dependsOn(['created_at'],
                    function (Text $field, NovaRequest $request, FormData $formData) {
                        logger(get_class($this).": Created at changed to ".var_export($formData->created_at, true));
                    }
                ),
```

If `Title` depends on `user`, `status` and `created_at` it should be `->dependsOn(['user', 'status', 'created_at'], ...)`

**Reply by kusab85** [reply]
I just wanted to have separate callback function for each field. 
Doesn't it work like that?

**Reply by crynobone** [reply]
No it doesn't work that way.

**Reply by kusab85** [reply]
Ok. THX.

**Comment by crynobone** [comment, marked as answer]
```php
Text::make('Title')
                ->rules('required')
                ->dependsOn(['user', 'status', 'created_at'],
                    function (Text $field, NovaRequest $request, FormData $formData) {
                        ray(get_class($this).": User changed to ".var_export($formData->user, true));
                        ray(get_class($this).": Status changed to ".var_export($formData->status, true));
                        ray(get_class($this).": Created at changed to ".var_export($formData->created_at, true));
                    }
                ),
```

![CleanShot 2022-09-02 at 09 33 01](https://user-images.githubusercontent.com/172966/188040139-421220aa-44a8-470e-8fed-8b955c2c7163.gif)

**Reply by kusab85** [reply]
In my case,  the root cause of the problem is that I'm using Ms IIS web server. 
In its default configuration, PHP CGI handler supports only **``GET, HEAD, POST``** http methods. 
```XML
<system.webServer>
  <handlers accessPolicy="Read, Script">
    <add name="PHP_via_FastCGI"
         path="*.php" 
         verb="GET,HEAD,POST" 
         modules="FastCgiModule" 
         scriptProcessor="C:\Program Files\PHP\v8.0\php-cgi.exe" 
         resourceType="Either" />
    <!-- ... -->
  </handlers>
<system.webServer>
```
Since ``dependsOn`` uses the **``PATCH``** method, this could not have worked.

I've changed config to ``verb="*"`` and everything works seamlessly!

**Reply by ProjectesFUAB** [reply]
marry me!!!
